test/test_cuda.py

```diff
@@ -1733,8 +1733,9 @@ def test_stream_event_nogil(self):
             t.join()
             tok = time.time()
 
-            self.assertGreater(tok - tik, 0.05)
-            self.assertLess(tok - tik, 0.1)
+            # the 5 ms room helps to avoid false alarm due to timing inaccuracy
+            self.assertGreater(tok - tik, 0.045)
+            self.assertLess(tok - tik, 0.095)
```

Comment:
This doesn't look like you made it roomier; it looks like you just shifted the entire range down 5ms.

Reply:
Yes, the expected run runtime should be a little bit higher than 50 ms without GIL. So, in theory, it should work if we use an even narrower range (e.g., 45ms - 55ms). 

I reduced the upper bound to 95ms to make sure the test fails if GIL is enforced. In theory, it should be a little bit larger than 100ms with GIL, but what Tongzhou reported suggests that it sometimes can go below 100ms.

Reply:
Let me update the comments.

Reply:
Have you seen 
https://github.com/pytorch/pytorch/issues/16271? That failure has a much lower time than 0.05. I think something is wrong with (one of?) these sync functions.

Reply:
That's definitely not expected. Let me investigate. Thanks for pointing it out!